lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -129,14 +129,42 @@ def generate_refresh_token
     end
 
     def generate_token
+      set_timestamps
+
       generator = Doorkeeper.configuration.access_token_generator.constantize
-      self.token = generator.generate(resource_owner_id: resource_owner_id,
-                                      scopes: scopes, application: application,
-                                      expires_in: expires_in)
+      self.token = generator.generate(
+        resource_owner_id: resource_owner_id,
+        scopes: scopes,
+        application: application,
+        expires_in: expires_in,
+        created_at: created_at
+      )
     rescue NoMethodError
       raise Errors::UnableToGenerateToken, "#{generator} does not respond to `.generate`."
     rescue NameError
       raise Errors::TokenGeneratorNotFound, "#{generator} not found"
     end
+
+    def set_timestamps
+      if created_at.nil?
+        self.created_at = current_time
+
+        if respond_to? :updated_at=
```

Comment:
> We can either play it safe and check for the existence of an updated_at field to ensure it gets set or we have to provide hooks for the ORM adapter to do the right thing here.

I don't think it makes sense for tokens to have an `updated_at` field, and I'd thus remove it from this PR. What hooks do you think external ORMs would need here?